parser.go

```diff
@@ -53,7 +53,7 @@ func (p *Parser) ParseArgs(args []string) error {
 	debugPrint("Kicking off parsing with args:", args)
 	err := p.parse(p, args, 0)
 	if err != nil {
-		return err
+		p.ShowHelpAndExit("Parse error: " + err.Error())
```

Comment:
I don't want to enable this because users already have the ability to check the error output from `ParseArgs`.  Also, this is in the path from the default `flaggy.Parse()` calls, which means it would potentially change the behavior for everyone.  This func currently does not assume it should invoke `ShowHelpAndExit()` and instead leaves that up to the user.